### PR TITLE
Add endpoints & models to start email verification

### LINKED ISSUE #4542: Implement email verification procedure for the email component
Taiga DH 684, business case "WMV", needs to be added to intranet still

To address mistakes in provided email addresses, we need to add an (optional) email verification flow for forms where the email address is crucial. The goals are:

* ensure that the user filling out the form does not enter wrong data and actually has access to the email account
* prevent forms from being submitted if the email is not verified

The verification process happens *during* the form filling, not after the form is submitted (different from typical user signup flows).

**User interaction**

1. User encounters an email account that requires verification, they enter `myname@example.com`
2. They click the button "send verification code" next to the email input field
3. a modal opens with a text input to enter the verification code + a descriptive text "that an email is being sent and it can take a couple of seconds for it to arrive. Please make sure to check your spam box yada yada..."
4. User enters verification code - we keep track of it in the submission in the SDK and send it in with the form step submit
5. if the code is wrong or not provided (modal dismissed), then we show a validation error with the email component that verification is required

**Imlementation**

- [x] Add an option to the email component `requireVerification` (can go in the `validation` nest obj I suppose)
- [x] Ensure verification button is displayed if a (valid) email is entered and the option is enabled
- [x] When verification is triggered, generate a code in the backend and save it along with the value of the email address, linked to the submission (note: we verify a particular email, not a component, consider changing the email value after verifying it!)
- [x] Ensure that a verification email is sent (add endpoints, templates, celery task...)
- [ ] Include backend validation of the verification code

### PR DESCRIPTION
**Pull request by sergei-maertens**
Closes #4542 partially

**Changes**

* Added data model + registration in the admin for email verifications
* Added API endoint to create an email verification
* Added documentation describing the flow

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

File: src/openforms/submissions/tests/test_admin_email_verifications.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+@disable_admin_mfa()
+class EmailVerificationAdminTests(WebTest):
+
+    def test_search_fields(self):
+        # smoke test that the search fields are correctly configured
+        user = SuperUserFactory.create()
+        url = reverse("admin:submissions_emailverification_changelist")
+        EmailVerificationFactory.create_batch(3)
+        EmailVerificationFactory.create(verified=True, email="openforms@opengem.nl")
```
Comment (vaszig) [inline src/openforms/submissions/tests/test_admin_email_verifications.py:19]: Maybe we should use another email for testing purposes?

File: src/openforms/submissions/tests/test_model_email_verification.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+from .factories import EmailVerificationFactory
+
+
+class EmailVerificationModelTests(SimpleTestCase):
+
+    @override_settings(LANGUAGE_CODE="en")
+    def test_string_representation(self):
+        with self.subTest("unverified"):
+            unverified = EmailVerificationFactory.build(
+                email="info@opengem.nl", verified_on=None, component_key="email"
```
Comment (vaszig) [inline src/openforms/submissions/tests/test_model_email_verification.py:12]: The same here?

File: src/openforms/submissions/models/email_verification.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        help_text=_("Key of the email component in the submission's form."),
+        validators=[variable_key_validator],
+    )
+    email = models.EmailField(
+        _("email address"),
+        help_text=_("The email address that is being verified."),
+    )
+    verification_code = models.CharField(
+        _("verification code"),
+        max_length=10,
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openforms/submissions/models/email_verification.py:61]: :thinking: yes, why

I'll change it :wink:

File: src/openforms/submissions/tests/test_admin_email_verifications.py (hunk 1 above)
Comment (sergei-maertens) [reply src/openforms/submissions/tests/test_admin_email_verifications.py:19]: what do you mean? I'm using (part of) this email to apply the search in the admin

Comment (sergei-maertens) [reply src/openforms/submissions/tests/test_admin_email_verifications.py:19]: discussed on slack :)

File: src/openforms/submissions/tests/test_model_email_verification.py (hunk 2 above)
Comment (sergei-maertens) [reply src/openforms/submissions/tests/test_model_email_verification.py:12]: discussed on slack :)

**Pull request merged**